File: python/collatz_conjecture.py
def steps(number: int) -> int:
    if number < 1:
        raise ValueError("Only positive integers are allowed")
    
    ct = 0
    while number != 1:
        if number % 2 == 0:
            number //=2
        else:
            number = 3 * number + 1
        ct += 1
    return ct

File: dart/collatz_conjecture.dart
class CollatzConjecture {
  int steps(int n) {
    if (n < 1) {
      throw ArgumentError('Only positive integers are allowed');
    }
    int ct = 0;
    while (n != 1) {
      if (n.isOdd) {
        n = n * 3 + 1;
      } else {
        n ~/= 2;
      }
      ct++;
    }
    return ct;
  }
}
